pkg/riskanalysis/cmd.go

```diff
@@ -92,80 +95,216 @@ func (opt *Options) Run() error {
+		},
+		Rows: rows,
+	}
+	fileName := filepath.Join(opt.JUnitDir, raReqLogFileName)
+	err := dataloader.WriteDataFile(fileName, dataFile)
 	if err != nil {
-		logrus.WithError(err).Error("Error writing risk analysis json artifact")
-		return nil
+		logrus.WithError(err).Warnf("unable to write data file: %s", fileName)
 	}
-	logrus.Infof("Successfully wrote: %s", outputFile)
+}
 
-	disruptionBytes := []byte(`{Backends: []}`)
-	da, err := runDisruptionAnalysis(opt, finalProwJobRun.ClusterData.JobType)
-	if err != nil {
-		logrus.WithError(err).Error("error running disruption analysis locally")
-		return nil
+// writeRATestResults writes the RA test results to an autodl file in the junit directory; errors abort with a log message
+func (opt *Options) writeRATestResults(analysisBytes []byte) {
+	var analysis struct {
+		Tests []struct {
+			Name string
+			Risk struct {
+				Level struct {
+					Name  string
+					Level int
+				}
+			}
+		}
+		OverallRisk struct {
+			Level struct {
+				Name  string
+				Level int
+			}
+		}
 	}
-	disruptionBytes, err = json.Marshal(da)
+	err := json.Unmarshal(analysisBytes, &analysis)
 	if err != nil {
-		logrus.WithError(err).Error("Error marshalling disruption results")
-		return nil
+		logrus.WithError(err).Error("Error unmarshalling risk analysis json")
+		return
 	}
 
-	// Write html file for spyglass
-	riskAnalysisHTMLTemplate := testdata.MustAsset("e2echart/test-risk-analysis.html")
-	html := bytes.ReplaceAll(riskAnalysisHTMLTemplate, []byte("TEST_RISK_ANALYSIS_SIPPY_URL_GOES_HERE"), []byte(sippyURL))
-	html = bytes.ReplaceAll(html, []byte("TEST_RISK_ANALYSIS_JSON_GOES_HERE"), riskAnalysisBytes)
-	html = bytes.ReplaceAll(html, []byte("TEST_DISRUPTION_ANALYSIS_JSON_GOES_HERE"), disruptionBytes)
-	path := filepath.Join(opt.JUnitDir, fmt.Sprintf("%s.html", "test-risk-analysis"))
-	if err := ioutil.WriteFile(path, html, 0644); err != nil {
-		logrus.WithError(err).Error("Error writing output file")
-		return nil
+	dataFile := dataloader.DataFile{
+		TableName: "risk_analysis_test_results",
+		Schema: map[string]dataloader.DataType{
+			"TestName":  dataloader.DataTypeString,
+			"RiskLevel": dataloader.DataTypeInteger,
+			"RiskName":  dataloader.DataTypeString,
+		},
+		Rows: []map[string]string{},
+	}
+	dataFile.Rows = append(dataFile.Rows, map[string]string{
+		"TestName":  "", // no test name means overall risk
+		"RiskLevel": fmt.Sprintf("%d", analysis.OverallRisk.Level.Level),
+		"RiskName":  analysis.OverallRisk.Level.Name,
+	})
+	for _, test := range analysis.Tests {
+		dataFile.Rows = append(dataFile.Rows, map[string]string{
+			"TestName":  test.Name,
+			"RiskLevel": fmt.Sprintf("%d", test.Risk.Level.Level),
+			"RiskName":  test.Risk.Level.Name,
+		})
+	}
+	err = dataloader.WriteDataFile(filepath.Join(opt.JUnitDir, raTestRiskFileName), dataFile)
+	if err != nil {
+		logrus.WithError(err).Error("Error writing risk analysis test results to autodl file")
```

Comment:
nit, but noticed you used the filename for the requests logs https://github.com/openshift/origin/blob/d30f5407d37771c8c9074049240960701f5b154f/pkg/riskanalysis/cmd.go#L254

Reply:
reproduced this pattern everywhere I think. a bit superfluous since i think the attached error includes the filename anyway, but why not.